packages/admin/blocks-admin/src/blocks/helpers/createCompositeBlockSelectField.tsx

```diff
@@ -25,5 +28,6 @@ export function createCompositeBlockSelectField<T extends string | number>({
                 <SelectField name="value" {...legacyFieldProps} {...fieldProps} options={options} />
             </BlocksFinalForm>
         ),
+        extractTextContents: (state) => extractTextContents?.(state) ?? [],
```

Comment:
For `createCompositeBlockSelectField` and `createCompositeBlockSwitchField` I override the default implementation to return an empty array.

For `createCompositeBlockTextField` I keep the default implementation (meaning it's returned as a string).